src/Microsoft.AspNet.Server.Kestrel/Http/SocketOutput.cs

```diff
@@ -91,6 +92,21 @@ public class SocketOutput : ISocketOutput
                 if (buffer.Count > 0)
                 {
                     var tail = ProducingStart();
+                    if (tail.IsDefault)
+                    {
+                        if (cancellationToken.CanBeCanceled)
+                        {
+                            return TaskUtilities.GetCancelledTask(
+                                cancellationToken.IsCancellationRequested ?
+                                cancellationToken :
+                                new CancellationToken(true));
```

Comment:
I guess that wasn't a yes or no question... It should be completed, not cancelled.